Standalone/PythonTests/Automated/benchmark_data_aggregator.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+"""
+Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+from argparse import ArgumentParser
+import json
+from pathlib import Path
+import time
+import copy
+import subprocess
+import requests
+
+from ly_test_tools.mars.filebeat_client import FilebeatClient
+
+class BenchmarkDataAggregator:
+    def __init__(self, workspace, logger, test_suite):
+        self.build_dir = Path(workspace.paths.build_directory())
+        self.results_dir = Path(workspace.paths.project()) / 'user' / 'Scripts' / 'PerformanceBenchmarks'
+        self.test_suite = test_suite
+        self.logger = logger
+        self.filebeat_client = FilebeatClient(logger)
+
+    def _process_benchmark(self, benchmark_dir):
+        benchmark_metadata = {}
+        frame_stats = { 'count': 0, 'totalTime': 0 }
+        pass_stats = {}
+
+        metadata_file = benchmark_dir / 'benchmark_metadata.json'
+        if metadata_file.exists():
+            data = json.loads(metadata_file.read_text())
+            benchmark_metadata = data['ClassData']
+        else:
+            self.logger.warn(f'Metadata file could not be found at {metadata_file}')
+
+        # this allows us to add additional data if necessary, e.g. frame_test_timestamps.json
+        is_timestamp_file = lambda file: file.name.startswith('frame') and file.name.endswith('_timestamps.json')
+
+        for file in benchmark_dir.iterdir():
+            if file.is_dir() or not is_timestamp_file(file):
+                continue
+
+            data = json.loads(file.read_text())
+            entries = data['ClassData']['timestampEntries']
+
+            frame_time = 0
+            for entry in entries:
+                name = entry['passName']
+                time_ns = entry['timestampResultInNanoseconds']
+                pass_entry = pass_stats.get(name, { 'totalTime': 0, 'maxTime': 0 })
+
+                pass_entry['maxTime'] = max(time_ns, pass_entry['maxTime'])
+                pass_entry['totalTime'] += time_ns
+                pass_stats[name] = pass_entry
+
+                frame_time += time_ns
+            frame_stats['totalTime'] += frame_time
+
+            frame_stats['maxTime'] = max(frame_time, frame_stats.get('maxTime', 0))
+            frame_stats['minTime'] = min(frame_time, frame_stats.get('minTime', frame_time))
+            frame_stats['count'] += 1
+
+        if frame_stats['count'] < 1:
+            self.logger.warn(f'No frame timestamp logs were found in {benchmark_dir}')
+
+        return benchmark_metadata, frame_stats, pass_stats
+
+    def _generate_payloads(self, benchmark_metadata, frame_stats, pass_stats):
```

Comment:
same here for docstrings, it gets hard to follow without knowing specifically the data type and value of each `param`

Reply:
UPDATE: This is solved now, won't let me resolve the comment though